### PR TITLE
Support requesting by ips and ports

### PR DESCRIPTION
**Pull request by supl**
This PR enhances RequestCoordinator to support requesting by IP and port addresses.

The Node part will be in another PR.

### CODE REVIEW

File: java/src/main/java/com/scalar/admin/AdminCommand.java [hunk 1]
```diff
@@ -59,7 +68,31 @@ public static void main(String[] args) {
 
   @Override
   public Integer call() {
-    RequestCoordinator coordinator = new RequestCoordinator(srvServiceUrl);
+    RequestCoordinator coordinator;
+
+    if (srvServiceUrl != null) {
```
Comment (kota2and3kan) [inline java/src/main/java/com/scalar/admin/AdminCommand.java:73]:
```suggestion
    if (srvServiceUrl != null && addresses != null) {
      throw new IllegalArgumentException(
          "Specify only either [--srv-service-url, -s] or [--addresses, -a].");
    } else if (srvServiceUrl != null) {
```
I think `--srv-service-url` and `--addresses` conflict.
So, I think it is better to check if both flags are specified and return an error if both are specified.
What do you think?

**Review by kota2and3kan — COMMENTED** [review]
Thank you for the update!
I left one suggestion. Please take a look!

File: java/src/main/java/com/scalar/admin/AdminCommand.java (hunk 1 above)
Comment (supl) [reply java/src/main/java/com/scalar/admin/AdminCommand.java:73]: @kota2and3kan 

It's a good idea.
I think this can prevent ambiguity.

**Review by kota2and3kan — APPROVED** [review]
LGTM! Thank you!

File: java/src/main/java/com/scalar/admin/RequestCoordinator.java [hunk 2]
```diff
@@ -52,25 +59,35 @@ public JsonObject checkPaused() {
 
   private Map<String, String> run(Function<AdminClient, Optional<String>> function) {
     // Assume that the list of addresses for unpause is the same as the one for pause.
-    List<SRVRecord> records = getApplicationIps(srvServiceUrl);
+    if (addresses == null) {
+      List<SRVRecord> records = getApplicationIps(srvServiceUrl);
+      addresses =
+          records.stream()
```
Comment (feeblefakie) [inline java/src/main/java/com/scalar/admin/RequestCoordinator.java:65]:
```suggestion
          getApplicationIps(srvServiceUrl);.stream()
```
No need for the `records` variable?

File: java/src/main/java/com/scalar/admin/RequestCoordinator.java [hunk 3]
```diff
@@ -52,25 +59,35 @@ public JsonObject checkPaused() {
...
+              .map(
+                  r -> {
+                    return new InetSocketAddress(r.getTarget().toString(true), r.getPort());
+                  })
```
Comment (brfrn169) [inline java/src/main/java/com/scalar/admin/RequestCoordinator.java:69]: Nit:
```suggestion
                  r -> new InetSocketAddress(r.getTarget().toString(true), r.getPort()))
```

**Review by brfrn169 — APPROVED** [review]
LGTM other than what I commented. Thank you!

File: java/src/main/java/com/scalar/admin/AdminCommand.java [hunk 4]
```diff
@@ -59,7 +68,34 @@ public static void main(String[] args) {
 
   @Override
   public Integer call() {
-    RequestCoordinator coordinator = new RequestCoordinator(srvServiceUrl);
+    RequestCoordinator coordinator;
+
+    if (srvServiceUrl != null && addresses != null) {
```
Comment (feeblefakie) [inline java/src/main/java/com/scalar/admin/AdminCommand.java:73]:
```suggestion
    if ((srvServiceUrl == null && addresses == null) || (srvServiceUrl != null && addresses != null)) {
```
to simplify the error handling (and to remove duplicate error messages)?

**Review by feeblefakie — COMMENTED** [review]
Overall looking good. Left minor suggestions. PTAL!

File: java/src/main/java/com/scalar/admin/RequestCoordinator.java (hunk 2 above)
Comment (supl) [reply java/src/main/java/com/scalar/admin/RequestCoordinator.java:65]: Yeah, it could be shorter.

**Comment by supl** [comment]
@feeblefakie 
Thank you for the advise.

I revise this PR according to Toshi and your feedbacks in https://github.com/scalar-labs/scalar-admin/pull/35/commits/b91638312c37ab66b59b5c41b2c927f637824462

I test it with my local ScalarDL cluster. It should be OK.

Please take a look 🙇

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**